grouper/fe/template_util.py

```diff
@@ -4,104 +4,78 @@
 from dateutil.relativedelta import relativedelta
 from jinja2 import Environment, PackageLoader, select_autoescape
 from pytz import UTC
-from six import string_types
 
 from grouper.fe.settings import settings
 
 if TYPE_CHECKING:
-    from typing import Any, Callable, Dict, Optional, Union
+    from typing import Any, Callable, Dict, Optional
 
+# Components of a relativedelta, in order from longest interval to shortest.
+DELTA_COMPONENTS = ["year", "month", "day", "hour", "minute", "second"]
 
-def _make_date_obj(date_obj):
-    # type: (Union[datetime, float, str]) -> datetime
-    """Given either a datetime object, float date/time in UTC unix epoch, or
-    string date/time in the form '%Y-%m-%d %H:%M:%S.%f' return a datetime
-    object."""
-    if isinstance(date_obj, float):
-        date_obj = datetime.fromtimestamp(date_obj, UTC)
 
-    if isinstance(date_obj, string_types):
-        date_obj = datetime.strptime(date_obj, "%Y-%m-%d %H:%M:%S.%f")
-        date_obj = date_obj.replace(tzinfo=UTC)
-
-    assert isinstance(date_obj, datetime)
+def highest_period_delta_str(delta):
+    # type: (relativedelta) -> Optional[str]
+    """Given a relativedelta, return a string version of only the longest interval."""
+    for component in DELTA_COMPONENTS:
+        value = getattr(delta, "{}s".format(component))
+        if value > 0:
```

Comment:
I think an explicit `if value < 0: return None` makes this a little easier to understand. Alternatively a note in the docstring that it returns None on negative (or zero) relativedeltas.